openmc/data/multipole.py

```diff
@@ -129,6 +143,600 @@ def _broaden_wmp_polynomials(E, dopp, n):
+    s = np.sqrt(energy)
+    test_s = np.sqrt(test_energy)
+
+    # inverse weighting is used for minimizing the relative deviation instead of
+    # absolute deviation in vector fitting
+    weight = 1.0/f
+
+    # avoid too large weights which will harm the fitting accuracy
+    min_cross_section = 1e-7
+    for i in range(nmt):
+        if np.all(ce_xs[i] <= min_cross_section):
+            weight[i] = 1.0
+        elif np.any(ce_xs[i] <= min_cross_section):
+            weight[i, ce_xs[i] <= min_cross_section] = \
+               max(weight[i, ce_xs[i] > min_cross_section])
+
+    # detect peaks (resonances) and determine VF order search range
+    peaks, _ = find_peaks(ce_xs[0] + ce_xs[1])
+    n_peaks = peaks.size
+    if orders is not None:
+        # make sure orders are even integers
+        orders = list(set([int(i/2)*2 for i in orders if i >= 2]))
+    else:
+        lowest_order = max(2, 2*n_peaks)
+        highest_order = max(200, 4*n_peaks)
+        orders = list(range(lowest_order, highest_order + 1, 2))
+
+    if log:
+        print("Found {} peaks".format(n_peaks))
+        print("Fitting orders from {} to {}".format(orders[0], orders[-1]))
+
+    # perform VF with increasing orders
+    found_ideal = False
+    n_discarded = 0 # for accelation, number of discarded searches
+    best_quality = best_ratio = -np.inf
+    for i, order in enumerate(orders):
+        if log:
+            print("Order={}({}/{})".format(order, i, len(orders)))
+        # initial guessed poles
+        poles_r = np.linspace(s[0], s[-1], order//2)
+        poles = poles_r + poles_r*0.01j
+        poles = np.sort(np.append(poles, np.conj(poles)))
+
+        found_better = False
+        # fitting iteration
+        for i_vf in range(n_vf_iter):
+            if log >= DETAILED_LOGGING:
+                print("VF iteration {}/{}".format(i_vf + 1, n_vf_iter))
+
+            # call vf
+            poles, residues, cf, f_fit, rms = vf.vectfit(f, s, poles, weight)
+
+            # convert real pole to conjugate pairs
+            n_real_poles = 0
+            new_poles = []
+            for p in poles:
+                p_r, p_i = np.real(p), np.imag(p)
+                if (s[0] <= p_r <= s[-1]) and p_i == 0.:
+                    new_poles += [p_r+p_r*0.01j, p_r-p_r*0.01j]
+                    n_real_poles += 1
+                else:
+                    new_poles += [p]
+            new_poles = np.array(new_poles)
+            # re-calculate residues if poles changed
+            if n_real_poles > 0:
+                if log >= DETAILED_LOGGING:
+                    print("  # real poles: {}".format(n_real_poles))
+                new_poles, residues, cf, f_fit, rms = \
+                      vf.vectfit(f, s, new_poles, weight, skip_pole=True)
+
+            # assess the result on test grid
+            test_xs = vf.evaluate(test_s, new_poles, residues) / test_energy
+            abserr = np.abs(test_xs - test_xs_ref)
+            relerr = abserr / test_xs_ref
```

Comment:
Since it looks we're expecting the possibility of nans being produced, probably makes sense to silence the warnings about them
```suggestion
            with np.errstate(invalid='ignore'):
                relerr = abserr / test_xs_ref
```

Reply:
ok